components/more_channels/more_channels.jsx

```diff
@@ -20,8 +19,11 @@ const CHANNELS_CHUNK_SIZE = 50;
 const CHANNELS_PER_PAGE = 50;
 const SEARCH_TIMEOUT_MILLISECONDS = 100;
 
-export default class MoreChannels extends React.Component {
+export default class MoreChannels extends React.PureComponent {
     static propTypes = {
+        channels: PropTypes.array.isRequired,
```

Comment:
This is /probably/ fine as is but if we're switching to a PureComponent then it might be a good idea to just pass in the channel IDs, not the actual channels, using a selector created with`createIdsSelector`. Then just use smaller components to get the actual channels. Otherwise any time a channel object changes (and they do change often) we'll be re-rendering.

Probably not a big deal here since this is for channels you're not already in and is only in a modal but something to think about

Reply:
Okay, I just retain it as `Component`.
I think we can address your recommendation separately, if necessary. Let me know if I need to create a ticket for that. Thanks!